array.c

```diff
@@ -3515,31 +3617,49 @@ rb_ary_sort_bang(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.sort                   -> new_ary
- *     ary.sort {|a, b| block}    -> new_ary
- *
- *  Returns a new array created by sorting +self+.
+ *    array.sort -> new_array
+ *    arrsy.sort {|a, b| ... } -> new_array
  *
- *  Comparisons for the sort will be done using the <code><=></code> operator
- *  or using an optional code block.
+ *  Returns a new \Array whose elements are those from +self+, sorted.
  *
- *  The block must implement a comparison between +a+ and +b+ and return
- *  an integer less than 0 when +b+ follows +a+, +0+ when +a+ and +b+
- *  are equivalent, or an integer greater than 0 when +a+ follows +b+.
+ *  ---
  *
- *  The result is not guaranteed to be stable.  When the comparison of two
- *  elements returns +0+, the order of the elements is unpredictable.
+ *  With no block, compares elements using operator <tt><=></tt>
+ *  (see Comparable):
+ *    a = 'abcde'.split('').shuffle
+ *    a # => ["e", "b", "d", "a", "c"]
+ *    a1 = a.sort
+ *    a1 # => ["a", "b", "c", "d", "e"]
  *
- *     ary = [ "d", "a", "e", "c", "b" ]
- *     ary.sort                     #=> ["a", "b", "c", "d", "e"]
- *     ary.sort {|a, b| b <=> a}    #=> ["e", "d", "c", "b", "a"]
+ *  ---
  *
- *  To produce the reverse order, the following can also be used
- *  (and may be faster):
+ *  With a block, calls the block with each element pair;
+ *  for each element pair +a+ and +b+, the block should return an integer:
+ *  - Negative when +b+ is to follow +a+.
+ *  - Zero when +a+ and +b+ are equivalent.
+ *  - Positive when +a+ is to follow +b+.
+ *
+ *  Example:
+ *    a = 'abcde'.split('').shuffle
+ *    a # => ["e", "b", "d", "a", "c"]
+ *    a1 = a.sort {|a, b| a <=> b }
+ *    a1 # => ["a", "b", "c", "d", "e"]
+ *    a2 = a.sort {|a, b| b <=> a }
+ *    a2 # => ["e", "d", "c", "b", "a"]
+ *
+ *  When the block returns zero, the order for +a+ and +b+ is indeterminate,
+ *  and may be unstable:
+ *    a = 'abcde'.split('').shuffle
+ *    a # => ["e", "b", "d", "a", "c"]
+ *    a1 = a.sort {|a, b| 0 }
+ *    a1 # =>  ["c", "e", "b", "d", "a"]
  *
- *    ary.sort.reverse!             #=> ["e", "d", "c", "b", "a"]
+ *  ---
  *
- *  See also Enumerable#sort_by.
```

Comment:
Any particular reason why we should remove this reference? I worry that by removing it may make it less discoverable

If you think it's reasonable to remove I'm OK with that

Reply:
I'm ok with keeping it.  It just didn't tell me anything I wanted to know re: Array#sort.

Reply:
I would justify it because constructing a comparison using the Comparable #<=> protocol can be difficult because you have to compare each field and get the return value (negative or positive) correct.

When using #sort_by ordering can be more straightforward to write and think about because you instead to select the field(s) you want to use for ordering and let Ruby do the comparison work based on those fields.

Reply:
Restored (near beginning).